x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/analysis/Analyzer.java

```diff
@@ -678,6 +684,45 @@ private List<Attribute> resolveUsingColumns(List<Attribute> cols, List<Attribute
             return resolved;
         }
 
+        private LogicalPlan resolveInsist(Insist insist, List<Attribute> childrenOutput, IndexResolution indexResolution) {
+            return insist.withAttributes(
+                insist.insistedAttributes().stream().map(a -> resolveInsistAttribute(a, childrenOutput, indexResolution)).toList()
+            );
+        }
+
+        private Attribute resolveInsistAttribute(Attribute attribute, List<Attribute> childrenOutput, IndexResolution indexResolution) {
+            assert attribute instanceof UnresolvedAttribute : "INSIST should be unresolved at this point";
```

Comment:
Any reason for adding the assertion?
The analysis rule only gets triggered on unresolved nodes